pkg/log/log.go

```diff
@@ -0,0 +1,46 @@
+package log
+
+import (
+	"os"
+
+	"go.uber.org/zap/zapcore"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var levelMap = map[string]zapcore.Level{
+	"debug":  zapcore.DebugLevel,
+	"info":   zapcore.InfoLevel,
+	"warn":   zapcore.WarnLevel,
+	"error":  zapcore.ErrorLevel,
+	"dpanic": zapcore.DPanicLevel,
+	"panic":  zapcore.PanicLevel,
+	"fatal":  zapcore.FatalLevel,
+}
+
+var logger = Logger()
+
+func Logger() logr.Logger {
+	l := os.Getenv("KUBEXIT_LOG_LEVEL")
+	level, ok := levelMap[l]
+	if !ok {
+		level = zapcore.InfoLevel
+	}
+
+	log.SetLogger(zap.New(zap.Level(level)))
```

Comment:
I haven't tested this, but afaik, zap.New returns a development logger which is not in json format, so you may need to use NewProduction instead https://github.com/uber-go/zap#quick-start

Reply:
Tested Info Level:
```
mcrouter {"level":"info","ts":1625553023.112102,"logger":"kubexit","msg":"Child Exited(0)"}
 mcrouter {"level":"info","ts":1625553023.112142,"logger":"kubexit","msg":"Updating tombstone:","path":"/graveyard/mcrouter"}
 mcrouter stream closed
 main {"level":"info","ts":1625553014.56097,"logger":"kubexit","msg":"Name:","name":"main"}
 main {"level":"info","ts":1625553014.5610745,"logger":"kubexit","msg":"Graveyard:","graveyard":"/graveyard"}
 main {"level":"info","ts":1625553014.5611012,"logger":"kubexit","msg":"Tombstone:","tombstone":"/graveyard/main"}
 main {"level":"info","ts":1625553014.5611215,"logger":"kubexit","msg":"Birth Deps:","birth deps":"mcrouter"}
 main {"level":"info","ts":1625553014.5611386,"logger":"kubexit","msg":"Death Deps: N/A"}
 main {"level":"info","ts":1625553014.5611424,"logger":"kubexit","msg":"Birth Timeout:","birth timeout":30}
 main {"level":"info","ts":1625553014.561163,"logger":"kubexit","msg":"Grace Period:","grace period":30}
 main {"level":"info","ts":1625553014.5611804,"logger":"kubexit","msg":"Pod Name:","pod name":"command-inject-fixtures-56khl-pqtfr"}
 main {"level":"info","ts":1625553014.5611854,"logger":"kubexit","msg":"Namespace:","namespace":"testmemcached-1625552852"}
 main {"level":"info","ts":1625553014.6600866,"logger":"kubexit","msg":"Watching pod updates..."}
```

Its in JSON format, I think we are good here.